### ISSUE TITLE
Create a calculate_team_stats() function

### ISSUE THREAD

**Issue by Josephhero** [nflverse/nflfastR#342]
Can I request that a calculate_team_stats() function be created, similar to the calculate_player_stats() function, that would calculate the official nfl tallies for team yards, TDs, points per drive, etc.?

**Comment by mrcaseb** [comment]
I am considering pfr for this. We could write scrapers for their team stats tables and put it all together in pfr_scrapR. What do you think @tanho63 and @guga31bb?

Some links:

- [OFF standard](https://www.pro-football-reference.com/years/2021/#all_team_stats) and tables below
- [OFF advanced](https://www.pro-football-reference.com/years/2021/advanced.htm): multiple tables and table tabs
- [DEF all](https://www.pro-football-reference.com/years/2021/opp.htm)

**Comment by tanho63** [comment]
I’m curious whether that is better or whether we should develop a true parallel to player stats (ie build up from pbp)

**Comment by guga31bb** [comment]
Yeah if we do this, it should probably be with pbp so people can subset the data before doing it (which isn't possible if we just scraped PFR)

**Comment by mrcaseb** [comment]
> Yeah if we do this, it should probably be with pbp so people can subset the data before doing it (which isn't possible if we just scraped PFR)

I agree this has some value over PFR. We'd miss out some pfr specific data by going the nflfastR pbp route tho. But we could at least have our own EP and WP data and add series conversion rates.

**Comment by mrcaseb** [comment]
I also wonder if we could go with this pseudo code

`calculate_player_stats(weekly = TRUE)` %>% summarise(team) %>% summarise(weeks if requested)

**Comment by tanho63** [comment]
hmm. if the target is the PFR standard offense tables I guess I don't see why not? We wouldn't have the series conversion rates though, nor redzone or drive averages by doing it that way. 

I'm sure we could reuse a lot of the code that prepares player_stats though. Briefly wondering if it should be written in data.table to make go faster - likely to be processing whole seasons worth of pbp with this function
